### PR TITLE
Baltaks

### PR DESCRIPTION
**Pull request by rzulian**
frreAction using handlers and disallowed nav if not PI

### CODE REVIEW

File: src/engine.spec.ts [hunk 1]
```diff
@@ -632,348 +632,392 @@ describe("Engine", () => {
...
-        p2 build m 0x-4
-        p1 build m -4x2
-        p2 booster booster3
-        p1 booster booster4
-      `);
+  });
 
-      const engine = new Engine(moves);
+  describe("free actions", () => {
+  it("should allow free action as first move after setupsetup is correct", () => {
```
Comment (coyotte508) [inline src/engine.spec.ts]: The indent is broken. It causes a huge commit !!

File: src/player.ts [hunk 2]
```diff
@@ -351,6 +351,10 @@ export default class Player extends EventEmitter {
     return this.data.gaiaPowerTokens() >= 4 && this.faction === Faction.Itars && this.data.hasPlanetaryInstitute();
   }
 
+  canUpgradeResearchBaltaks( field: ResearchField): boolean {
+    return (this.data.hasPlanetaryInstitute() && field === ResearchField.Navigation) || field !== ResearchField.Navigation;
```
Comment (coyotte508) [inline src/player.ts]: no need for parentheses around the return

File: src/engine.spec.ts (hunk 1 above)
Comment (rzulian) [reply src/engine.spec.ts]: I'll resubmit one with the right indentation. Why isn't lint correcting this?

File: src/player.ts (hunk 2 above)
Comment (rzulian) [reply src/player.ts]: ok

File: src/engine.spec.ts (hunk 1 above)
Comment (rzulian) [reply src/engine.spec.ts]: I've only added the Baltaks part (describe("Baltaks")

Comment (rzulian) [reply src/engine.spec.ts]: Working on the tests... please wait

**Comment by coyotte508** [comment]
The indentation is fixed, but not for the content in parseMoves

File: src/player-data.ts [hunk 3]
```diff
@@ -119,7 +120,7 @@ export default class PlayerData extends EventEmitter {
       case Resource.ChargePower: count > 0 ? this.chargePower(count) : this.spendPower(-count); return;
       case Resource.Range: this.range += count; return;
       case Resource.TemporaryRange: this.temporaryRange += count; return;
-      case Resource.GaiaFormer: this.gaiaformers += count; return;
+      case Resource.GaiaFormer: count > 0 ? this.gaiaformers += count : this.gaiaformersInGaia -= count; return;
```
Comment (coyotte508) [inline src/player-data.ts]: `this.gaiaformers += count` should be done always. Then, if count < 0, ` this.gaiaformersInGaia -= count`.

File: src/player.ts [hunk 4]
```diff
@@ -422,6 +426,8 @@ export default class Player extends EventEmitter {
     }
 
     this.data.power.gaia = 0;
+    this.data.gaiaformers += this.data.gaiaformersInGaia;
```
Comment (coyotte508) [inline src/player.ts]: I don't think it should be there.

Comment (coyotte508) [reply src/player.ts]: It's not decreased in `gainReward`, and `hasResource` also expects it to not be decreased

**Comment by coyotte508** [comment]
I manually merged the changes

**Pull request closed without merging**